docs/src/views/Architecture.tsx

```diff
@@ -0,0 +1,224 @@
+      object, which has three major responsibilities:
+    </p>
+    <ul>
+      <li>
+        Define <code>aria-*</code> attributes for the standard components (e.g. Image, DropDown,
+        Dialog, etc), based on their state (e.g. active, disabled, expanded, etc).
+      </li>
+      <li>
+        Define the correct element <code>role</code>, if the HTML element tag differ from standard
+        (e.g. use <code>&lt;span&rt;</code> instead of <code>&lt;button&rt;</code>) to represent the
+        button element.
+      </li>
+      <li>
+        Define the keyboard navigation and focus handling for the composed components (e.g. List,
+        Grid, Radio Group, etc).
+      </li>
+    </ul>
+    <p>
+      Library provides the set of accessibility behaviors and keyboard handlers for the components
+      and allows to create and inject your own, if it is necessary for some custom components. In
+      most cases it should be enough to use the behaviors provided in the library, but, of course,
+      you can provide your own or extend existing behaviors.
+    </p>
+    <p>
+      In addition to behaviors and keyboard handlers, the toolkit also provides FocusZone component,
+      which can be used to add focus handling for the wrapped component. There is also FocusTrap
+      component, which allows to trap the focus inside some area, which is useful for such a
+      component as Dialog.
+    </p>
+
+    <Header as="h3">Theming</Header>
+    <p>
+      You can inject different themes into Stardust components (e.g. Light, Dark or High Contrast).
+      To achieve this, we use CSS in JS technique, which is provided by{' '}
+      <a href="http://fela.js.org/" target="_blank">
+        Fela library
+      </a>.
+    </p>
+    <p>
+      Each component has its own set of themes defined. Component logic handles theme switching and
+      style overriding by providing ability to set <strong>variables</strong> or override component
+      (or its children) styles using <code>styles</code> property.
+    </p>
+    <p>
+      Component styling is managed by the theme provider, which has to wrap components to provide
+      default theme and theme switching. Providers can be nested, this is necessary in case if you
+      need to use different theme in children and parent components.
+    </p>
+    <p>
+      Additional info about how to use themes or override styles can be found in the{' '}
+      <a href="themeing">Theming guide</a>.
+    </p>
+
+    <Header as="h3">UIComponent</Header>
+    <p>
+      The <code>UIComponent</code> is the base component, which handles component default
+      properties, accessibility navigation and focus handling and renders the component.
+    </p>
+    <p>
+      We recommend to use <code>UIComponent</code> instead of <code>React.Component</code> when you
+      create your own primitive or composed component, to leverage ability of injecting
```

Comment:
We **do not** recommend using UIComponent, we'd actually like to get rid of it if possible.  The API for consumers to create their own components is still underway, however, we hope that it does not involved extending our component.  It will likely be something like `createComponent()` or `renderComponent()`.

We do not want to spread knowledge of UIComponent.